src/material-experimental/mdc-checkbox/checkbox.ts

```diff
@@ -0,0 +1,374 @@
+
+  /** Event emitted when the checkbox's `indeterminate` value changes. */
+  @Output() readonly indeterminateChange: EventEmitter<boolean> = new EventEmitter<boolean>();
+
+  /** The root element for the `MDCCheckbox`. */
+  @ViewChild('checkbox') _checkbox: ElementRef<HTMLElement>;
+
+  /** The native input element. */
+  @ViewChild('nativeCheckbox') _nativeCheckbox: ElementRef<HTMLInputElement>;
+
+  /** The native label element. */
+  @ViewChild('label') _label: ElementRef<HTMLElement>;
+
+  /** Returns the unique id for the visual hidden input. */
+  get inputId(): string {
+    return `${this.id || this._uniqueId}-input`;
+  }
+
+  /** The `MDCCheckboxFoundation` instance for this checkbox. */
+  _checkboxFoundation: MDCCheckboxFoundation;
+
+  /** The set of classes that should be applied to the native input. */
+  _classes = new Map([['mdc-checkbox__native-control', true]]);
+
+  /** ControlValueAccessor onChange */
+  private _cvaOnChange = (_: boolean) => {};
+
+  /** ControlValueAccessor onTouch */
+  private _cvaOnTouch = () => {};
+
+  /** The ripple renderer for this checkbox. */
+  private _rippleRenderer: RippleRenderer;
+
+  /**
+   * A list of attributes that should not be modified by `MDCFoundation` classes.
+   *
+   * MDC uses animation events to determine when to update `aria-checked` which is unreliable.
+   * Therefore we disable it and handle it ourselves.
+   */
+  private _attrBlacklist = new Set(['aria-checked']);
+
+  /** The `MDCCheckboxAdapter` instance for this checkbox. */
+  private _checkboxAdapter: MDCCheckboxAdapter = {
+    addClass: (className) => this._setClass(className, true),
+    removeClass: (className) => this._setClass(className, false),
+    forceLayout: () => this._platform.isBrowser && this._checkbox.nativeElement.offsetWidth,
+    hasNativeControl: () => !!this._nativeCheckbox.nativeElement,
+    isAttachedToDOM: () => !!this._checkbox.nativeElement.parentNode,
+    isChecked: () => this.checked,
+    isIndeterminate: () => this.indeterminate,
+    removeNativeControlAttr:
+        (attr) => {
+          if (!this._attrBlacklist.has(attr)) {
+            this._nativeCheckbox.nativeElement.removeAttribute(attr);
+          }
+        },
+    setNativeControlAttr:
+        (attr, value) => {
+          if (!this._attrBlacklist.has(attr)) {
+            this._nativeCheckbox.nativeElement.setAttribute(attr, value);
+          }
+        },
+    setNativeControlDisabled: (disabled) => this.disabled = disabled,
+  };
+
+  constructor(
+      private _changeDetectorRef: ChangeDetectorRef,
+      private _platform: Platform,
+      private _ngZone: NgZone,
+      @Attribute('tabindex') tabIndex: string,
+      @Optional() @Inject(MAT_CHECKBOX_CLICK_ACTION) private _clickAction: MatCheckboxClickAction,
+      @Optional() @Inject(ANIMATION_MODULE_TYPE) public _animationMode?: string) {
+    this.tabIndex = parseInt(tabIndex) || 0;
```

Comment:
Since you're doing this, you can get rid of the initializer for `tabIndex`. AFAIK it should generate slightly less JS code.